src/main/java/io/appium/mitmproxy/MitmproxyJava.java

```diff
@@ -10,99 +12,120 @@
 import java.io.InputStream;
 import java.net.InetSocketAddress;
 import java.net.Socket;
-import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeoutException;
 import java.util.function.Function;
 
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+
+@Slf4j
 public class MitmproxyJava {
 
-    private String mitmproxyPath;
+    private static final String LOCALHOST_IP = "127.0.0.1";
+    private static final int WEBSOCKET_PORT = 8765;
+
+    private final String mitmproxyPath;
+
+    private final Function<InterceptedMessage, InterceptedMessage> messageInterceptor;
+
+    private final int proxyPort;
+
     private MitmproxyServer server;
+
+    private final List<String> extraMitmdumpParams;
+
     private Future<ProcessResult> mitmproxyProcess;
-    public static final int WEBSOCKET_PORT = 8765;
 
-    public MitmproxyJava(String mitmproxyPath, Function<InterceptedMessage, InterceptedMessage> messageInterceptor) {
+    public MitmproxyJava(String mitmproxyPath, Function<InterceptedMessage, InterceptedMessage> messageInterceptor, int proxyPort, List<String> extraMitmdumpParams) {
         this.mitmproxyPath = mitmproxyPath;
-        server = new MitmproxyServer(new InetSocketAddress("localhost", WEBSOCKET_PORT), messageInterceptor);
-        server.start();
+        this.messageInterceptor = messageInterceptor;
+        this.proxyPort = proxyPort;
+        this.extraMitmdumpParams = extraMitmdumpParams;
     }
 
-    public void start() throws IOException, TimeoutException, URISyntaxException {
-        System.out.println("starting mitmproxy on port 8080");
+    public MitmproxyJava(String mitmproxyPath, Function<InterceptedMessage, InterceptedMessage> messageInterceptor) {
+        this(mitmproxyPath, messageInterceptor, 8080, null);
+    }
+
+    public void start() throws IOException, TimeoutException {
```

Comment:
I think it might be useful if start could return the self instance

Reply:
same for other public methods

Reply:
there is only stop() left :)